stock_move_common_dest/models/stock_move.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, fields, models
+from odoo.osv.expression import FALSE_DOMAIN
+
+
+class StockMove(models.Model):
+
+    _inherit = "stock.move"
+
+    common_dest_move_ids = fields.Many2many(
+        "stock.move",
+        compute="_compute_common_dest_move_ids",
+        search="_search_compute_dest_move_ids",
+        help="All the stock moves having a chained destination move sharing the"
+        " same picking as the actual move's destination move",
+    )
+
+    def _common_dest_move_query(self):
+        sql = """SELECT smmr.move_orig_id move_id
+            , array_agg(smmr2.move_orig_id) common_move_dest_ids
+            FROM stock_move_move_rel smmr
+            , stock_move sm_dest
+            , stock_picking sp
+            , stock_move sm_pick
+            , stock_move_move_rel smmr2
+            WHERE smmr.move_dest_id = sm_dest.id
+            AND sm_dest.picking_id = sp.id
+            AND sp.id = sm_pick.picking_id
+            AND sm_pick.id = smmr2.move_dest_id
+            AND smmr.move_orig_id != smmr2.move_orig_id
+            AND smmr.move_orig_id IN %s
+            GROUP BY smmr.move_orig_id;
+        """
+        return sql
+
+    @api.depends(
+        "move_dest_ids",
+        "move_dest_ids.picking_id",
+        "move_dest_ids.picking_id.move_lines",
+        "move_dest_ids.picking_id.move_lines.move_orig_ids",
+    )
+    def _compute_common_dest_move_ids(self):
+        sql = self._common_dest_move_query()
+        self.env.cr.execute(sql, (tuple(self.ids),))
```

Comment:
Don't we need a `flush()` here? Probably a bug in 13.0 though...

Reply:
@grindtildeath It's a SELECT query, why would we need a flush?

Reply:
@gurneyalex I just looked a bit more into Odoo. documentation :sweat_smile: 

On one hand, we have the `flush` function description that is: "Process all the pending computations (on all models), and flush all the pending updates to the database." (https://www.odoo.com/documentation/14.0/reference/orm.html#odoo.models.Model.flush)
So even if we have a `SELECT` statement, I tend to think we need to flush the models/fields that are used in the query to make sure we'll have the same result than we'd have using ORM methods.

On the other hand, we have "SQL Execution" (https://www.odoo.com/documentation/14.0/reference/orm.html#sql-execution) that states the following: "Because models use the same cursor and the Environment holds various caches, these caches must be invalidated when altering the database in raw SQL, or further uses of models may become incoherent. It is necessary to clear caches when using CREATE, UPDATE or DELETE in SQL, but not SELECT (which simply reads the database)."
We should also remember this was written before v13.0 and maybe wasn't updated properly (https://www.odoo.com/documentation/12.0/reference/orm.html#low-level-sql) 

Then if we look into the code, we can see a few examples of code in Odoo where `flush` is used before `SELECT` statements:
https://github.com/odoo/odoo/blob/14.0/addons/account/models/account_account.py#L104-L105
https://github.com/odoo/odoo/blob/14.0/addons/account/models/account_account.py#L125-L132
https://github.com/odoo/odoo/blob/14.0/addons/product_margin/models/product_product.py#L106-L108

We can probably also see others where there's no flush though...
https://github.com/odoo/odoo/blob/14.0/addons/point_of_sale/models/pos_order.py#L895

Finally, as I understand it, v13.0 brought some improvements in ORM, where transactions are not necessarily written directly in the DB but left pending in memory to reduce the amount of interactions with the DB by regrouping updates.
Therefore, anytime we _bypass_ the ORM (i.e. executing SQL through DB cursor), we should make sure that the values still in the ORM cache are reflected in the DB or we risk data inconsistencies, and that's why Odoo provides `flush` method.

The explanations from Odoo experience seem to confirm my assumption: https://youtu.be/EBvuvyIJ1d0 (around 24-25 mins)
Too bad Odoo doesn't update its documentation properly, and this wasn't described in OCA migration guidelines neither....

Reply:
so yes, you are right, I read too fast (flush **before** makng the query)